File: src/main/java/SimpleServerApp.java
import com.esotericsoftware.kryonet.Connection;
import com.esotericsoftware.kryonet.Listener;
import com.esotericsoftware.kryonet.Server;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import java.io.IOException;


public class SimpleServerApp {
    public static void main(String[] args) {
        Logger logger = LogManager.getLogger(SimpleServerApp.class);
        logger.info("asdsadsa");
        Server server = new Server();
        NetworkUtil.register(server.getKryo());
        server.start();

        try {
            server.bind(54555, 54777);
            server.addListener(new Listener() {
                @Override
                public void received(Connection connection, Object object) {
                    if (object instanceof SimpleRequest) {
                        SimpleRequest sq = (SimpleRequest) object;
                        logger.info("Got some info from the socket: sq.getText()");
                        try {
                            SimpleResponse sr = new SimpleResponse(Integer.parseInt(sq.getText()));
                            connection.sendTCP(sr);
                        } catch (NumberFormatException e) {
                            logger.warn("Exception while trying to parse request text as number.");
                            logger.warn(e.getMessage());
                        }

                    }
                }
            });
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: src/main/kotlin/SimpleServerApp.kt
import com.esotericsoftware.kryonet.Connection
import com.esotericsoftware.kryonet.Listener
import com.esotericsoftware.kryonet.Server
import org.apache.logging.log4j.LogManager
import java.io.IOException

object SimpleServerApp {
    @JvmStatic
    fun main(args: Array<String>) {
        val logger = LogManager.getLogger(SimpleServerApp::class.java)
        logger.info("asdsadsa")
        val server = Server()
        NetworkUtil.register(server.kryo)
        server.start()

        try {
            server.bind(54555, 54777)
            server.addListener(object : Listener() {
                override fun received(connection: Connection?, obj: Any?) {
                    if (obj is SimpleRequest) {
                        logger.info("Got some info from the socket: sq.getText()")
                        try {
                            val sr = SimpleResponse(Integer.parseInt(obj.text).toDouble())
                            connection!!.sendTCP(sr)
                        } catch (e: NumberFormatException) {
                            logger.warn("Exception while trying to parse request text as number.")
                            logger.warn(e.message)
                        }

                    }
                }
            })
        } catch (e: IOException) {
            e.printStackTrace()
        }


    }
}